### ISSUE TITLE
Process to merge or ticket or comment on bot BOM PRs

### ISSUE THREAD

**Issue by robrap** [edx/edx-arch-experiments#826]
Using this [github search for open BOM PRs](https://github.com/pulls?q=is%3Aopen+is%3Apr+archived%3Afalse+repo%3Aopenedx%2Fauth-backends+repo%3Aopenedx%2Fedx-django-utils+repo%3Aopenedx%2Fedx-drf-extensions+repo%3Aopenedx%2Fedx-repo-health+repo%3Aopenedx%2Fevent-bus-kafka+repo%3Aopenedx%2Fhelp-tokens+sort%3Acreated-asc), you can see many bot PRs, some of which are really old. We'd like a process to clarify the status, and to help ensure that arbi-bom can get these closed.

Acceptance Criteria:
- [ ] Document process that makes status of open bot PRs more clear, especially older PRs, with a plan for how to get them merged.
- [ ] Initiate process on some PRs to prove out the process.

Notes:
- Are there PRs that should be notifying arbi-bom, but they are not?
- How can we tell if a PR is still open because it was never seen, or if it was seen, but not prioritized, or it has some other blocking issue that will take some time to fix.
  - Can PR comments or labels resolve this? Other?

**Comment by UsamaSadiq** [comment]
## Analysis of Pending Pull Requests

A comprehensive review of the pending pull requests (PRs) indicates that they predominantly fall into three distinct categories:

- **Routine Upgrades:** Encompassing standard dependency updates (e.g., requirements upgrade PRs, Dependabot-initiated PRs).
- **One-Time Upgrade Efforts:** Involving targeted modifications such as refactoring of `setup.py` and removal of the Codecov package.
- **R&D Initiatives:** Representing experimental and exploratory projects (e.g., the repository health dashboard and event-bus-kafka integration).

## Proposed Review Process Enhancement

Following detailed internal discussions, the team recommends instituting a dedicated evaluation process to manage the review and integration of these PRs. The key elements of the proposal are as follows:

- **Scheduled Review Day:**  
  Designate a specific day—preferably Wednesday, due to a lower frequency of daily meetings—for a comprehensive review session of all outstanding PRs within the arbi-bom and arch-bom repositories.

- **Integration with On-Call Rotation:**  
  The arbi-bom team currently operates an on-call rotation on a weekly basis. Allocating a few hours on the designated day for routine upgrade reviews will ensure systematic processing and prevent any PR from being overlooked.

- **Enhanced Handover Protocol:**  
  The on-call team member is tasked with providing detailed comments on each reviewed PR. This practice will ensure that context and critical information are effectively communicated during the transition to the next on-call engineer, thereby maintaining continuity and mitigating the risk of context loss.

## Suggested Implementation Steps

1. **Identify a Fixed Review Slot:**  
   Confirm Wednesday as the standard review day and schedule a recurring time slot specifically for PR evaluation.

2. **Integrate with Existing On-Call Workflow:**  
   Amend the current on-call process to include the PR review session. Ensure that the on-call member is aware of this additional responsibility.

3. **Establish a Handover Documentation Standard:**  
   Develop a template or checklist for the on-call member to use when documenting review comments and context transfer, enhancing the clarity and utility of the handover process.

4. **Monitor and Iterate:**  
   After implementation, continuously monitor the efficacy of the process and solicit feedback from the team to refine the approach.

This structured and systematic approach aims to streamline PR management, ensuring timely integration of updates and fostering effective knowledge transfer within the team.

**Comment by UsamaSadiq** [comment]
Published the draft of the arbi-bom On-Call process confluence [document](https://2u-internal.atlassian.net/wiki/x/AYCADg) with the updated steps regarding this scheduled review for the on-call person.

**Comment by UsamaSadiq** [comment]
@robrap Now that the team is back from vacations, could you have a final look at the suggested solution to handle the bot BOM PRs. If you don't have any further suggestions, I'll mark it as our final approach and close the issue to implement this within the team processes.

**Comment by robrap** [comment]
@UsamaSadiq: The one change I'd love to see is to clarify that when a PR is being reviewed (or triaged), it should result in either:
1. Merging, or
2. Adding a PR comment that clarifies status. For example: "This PR will take time to merge due to ___. Here is a follow-up ticket for getting it merged." Or, anything of that sort, so we know what PRs are sitting around because they've never been seen, vs those that will take some work to get over the line.

Thoughts?

**Comment by UsamaSadiq** [comment]
> [@UsamaSadiq](https://github.com/UsamaSadiq): The one change I'd love to see is to clarify that when a PR is being reviewed (or triaged), it should result in either:
> 
> 1. Merging, or
> 2. Adding a PR comment that clarifies status. For example: "This PR will take time to merge due to ___. Here is a follow-up ticket for getting it merged." Or, anything of that sort, so we know what PRs are sitting around because they've never been seen, vs those that will take some work to get over the line.
> 
> Thoughts?

Yes, Ideally the on-call person will be trying to merge the PRs as soon as possible after reviewing the changes. In case the PR is blocked on something, a detailed comment or a follow-up issue on the `arbi-bom` project will be created for the team to handle. I'll also clarify this point in the confluence document as well.

**Comment by robrap** [comment]
Thank you. Once clarified, feel free to close this and experiment away.

**Comment by UsamaSadiq** [comment]
Updated the confluence document. Team will start practicing this framework starting from the current week.
